models/user_setting.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
```

Comment:
How about move this into `models/user/setting.go` so that we don't need move it again in future PRs.

Reply:
updated. Unit tests are now failing, but can't look into why. Do you have ideas on why it could fail?

Reply:
Hi, a `main_test.go` file should be added in the sub package and below codes should be in that file:

```go
func TestMain(m *testing.M) {
	db.MainTest(m, filepath.Join("..", ".."))
}
```

Reply:
If you do not need to load these test fixtures, you can use this:

```
db.MainTest(m, filepath.Join("..", ".."), "")
```

Set the `fixureFiles` to an empty string, like this: https://github.com/go-gitea/gitea/blob/main/models/issues/main_test.go

I just patched the `MainTest` before. 😊

Reply:
Thanks for the help :) I merged from `main` and now have those changes.